src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -41,10 +42,21 @@
  * @author lwahlmeier
  */
 public class NoThreadSocketExecuter extends AbstractService implements SocketExecuterInterface {
+  public static final int WATCHDOG_CLEANUP_TIME = 30000;
+  
   private final NoThreadScheduler scheduler = new NoThreadScheduler();
   private final ConcurrentHashMap<SocketChannel, Client> clients = new ConcurrentHashMap<SocketChannel, Client>();
   private final ConcurrentHashMap<SelectableChannel, Server> servers = new ConcurrentHashMap<SelectableChannel, Server>();
   private final SocketExecuterByteStats stats = new SocketExecuterByteStats();
+  private final WatchdogCache dogCache = new WatchdogCache(scheduler);
+  private final Runnable watchDogCleanup = new Runnable() {
+    @Override
+    public void run() {
+      if(isRunning()) {
+        dogCache.cleanup();
+        scheduler.schedule(this, WATCHDOG_CLEANUP_TIME);
```

Comment:
If cleanup somehow throws an exception this will prevent future cleanups from running.  I know it should not, but still better safe than sorry.